addons/website_livechat/static/src/components/discuss/discuss.xml

```diff
@@ -1,10 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <templates xml:space="preserve">
-    <t t-inherit="mail.Discuss.content" t-inherit-mode="extension">
-        <xpath expr="//*[hasclass('o_Discuss_thread')]" position="before">
-            <t t-if="discuss.thread.visitor">
+    <t t-inherit="mail.ThreadView" t-inherit-mode="extension">
+        <xpath expr="//t[@t-if='threadView.topbar']" position="after">
+            <t t-if="threadView.thread.visitor and threadView.topbar">
```

Comment:
Presence of visitor should not necessarily depend on presence of topbar.

I would suggest to introduce a new computed field `hasVisitorBanned` in `threadView` for this condition, that checks
1. thread has visitor
2. thread viewer has the appropriate type (at least `threadViewer.discuss`, and maybe `discussPublicView`?)